sematic/db/queries.py

```diff
@@ -311,6 +330,42 @@ def get_jobs_by_run_id(run_id: str, kind: JobKindString = JobKind.run) -> List[J
         )
 
 
+def count_jobs_by_run_id(run_id: str, kind: JobKindString = JobKind.run) -> int:
+    """Count jobs from the DB by source run id.
+
+    Parameters
+    ----------
+    run_id:
+        The id of the run to get jobs for
+    kind:
+        The kind of jobs to get. Can be either "run" to get
+        jobs for the run itself, or "resolution" to get jobs
+        for the resolution associated with the run.
+
+    Returns
+    -------
+    The number of jobs associated with the run.
+    """
+    with db().get_session() as session:
+        return (
+            session.query(Job)
+            .filter(Job.run_id == run_id)
+            .filter(Job.kind == kind)
+            .count()
+        )
+
+
+# TODO: Remove this function
+# https://github.com/sematic-ai/sematic/issues/710
+def run_has_legacy_jobs(run_id: str) -> bool:
+    with db().get_session() as session:
+        statement = sqlalchemy.text(
+            "SELECT external_jobs_json FROM runs WHERE id=:run_id"
```

Comment:
No, because the `Run` object no longer has an `external_jobs_json` field. I removed it because you requested that I do so ;-) . And the query just gets the raw json; I'm not sure there's a Postgres/SQLite compatible way to get the count of the json contents, and doing a `len` is easy enough. Plus this query need not be perfect; it's used only in one place and is only temporary.